pallets/xor-fee/src/lib.rs

```diff
@@ -714,6 +785,22 @@ pub mod pallet {
             Self::deposit_event(Event::WeightToFeeMultiplierUpdated(new_multiplier));
             Ok(().into())
         }
+
+        #[allow(unused_variables)]
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::set_fee_update_period())]
+        pub fn set_fee_update_period(
+            origin: OriginFor<T>,
+            next_block_number: Option<<T as frame_system::Config>::BlockNumber>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            #[cfg(feature = "ready-to-test")] // Dynamic fee
+            {
+                <NextUpdateBlock<T>>::put(next_block_number);
```

Comment:
Probably you misunderstand the task. `BlocksToUpdate` should not be a config param, it shoud be a storage.
And this extrinsic should not just update the next block where hook is triggered, it should update this period.